optimade/server/middleware.py

```diff
@@ -3,20 +3,23 @@
 from starlette.requests import Request
 from starlette.responses import RedirectResponse
 
+from .routers import ENTRY_COLLECTIONS
+
 
 class RedirectSlashedURLs(BaseHTTPMiddleware):
     async def dispatch(self, request: Request, call_next):
         """Redirect URL requests ending with a slash to non-slashed URLs
 
-        E.g., `http://example.org/info/` -> `http://example.org/info`
+        E.g., `http://example.org/optimade/v0/info/` -> `http://example.org/optimade/v0/info`
         """
-        if request.scope["path"].endswith("/"):
+        if request.scope["path"].endswith("/") and any(
+            request.scope["path"].endswith(f"{endpoint}/")
+            for endpoint in list(ENTRY_COLLECTIONS.keys())
```

Comment:
This doesn't catch `/info/`. See if you can use the `app`'s routes instead or something else?
Or maybe it's time we made a set/list of all available endpoints - at the moment it's kind of hard-coded into the `info` router ...

Reply:
Well spotted, I didn't include `info` here as I thought the redirect was only necessary for endpoints that have this SingleEntry/ManyEntry behaviour. After testing it looks like `/info/` (as opposed to `/info`) currently 404s, so we should add it here.